janus_core/src/test_util/kubernetes.rs

```diff
@@ -90,6 +145,22 @@ impl Drop for EphemeralCluster {
     }
 }
 
+/// A guard on a running `kubectl port-forward` process. The process will be killed and reaped when
+/// the guard is dropped.
+pub struct PortForwardGuard {
+    child: Child,
+}
+
+impl Drop for PortForwardGuard {
+    fn drop(&mut self) {
+        debug!(?self.child, "dropping port forward");
+        // We kill and reap child `kubectl` processes here to ensure that we don't create zombies on
+        // test failures. This is normally a questionable practice, but acceptable since this is
+        // only used in test code.
```

Comment:
I think waiting on the process post-`kill()` is reasonable: other than being more accurate to the comment, I think it would also be slightly cleaner if dropping the value didn't complete until the port forward (i.e. process) is actually gone, rather than just "signalled to die soon in a way that it can't(?) override".